scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2198,7 +2198,21 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       val lpPos = auto.startTokenPos
       val args = checkNoTripleDots(argumentExprs(location))
       val rpPos = auto.endTokenPos
-      @inline def addPos(body: Term): Term = atPosWithBody(lpPos, body, rpPos)
+      @inline def addPos(body: Term): Term = {
+        body match {
+          // For `Term.Ref`, avoid using autoPos to exclude parens tokens in Term.ApplyInfix.args.
+          // For `a f (b)`, autoPos include `(b)` for ApplyInfix.args position.
+          // https://github.com/scalacenter/scalafix/issues/1594
+          case _: Term.Ref =>
+            val (start, end) = body.origin match {
+              case Origin.None => (lpPos, rpPos)
+              case origin: Origin.Parsed => (origin.pos.start, origin.pos.end - 1)
+            }
+            atPosWithBody(start, body, end)
+          case _ =>
+            atPosWithBody(lpPos, body, rpPos)
+        }
+      }
```

Comment:
The `body: Term` is the argument part of `ApplyInfix` (e.g. for `a f b`, `b` part will come here).

I'm not sure `Term.Tuple` and `Term.ApplyInfix` are the only nodes that should be surrounded by parentheses. For example, `a f (if (x) 1 else 2)`,  we will have `Term.If` and parens have to be included.

I believe it is safer to

- `Term.Ref` seems like not need to be surrounded by parens, so employ the position of `body.Term`.
- Otherwise, it might be necessary to be surrounded by parens, so leave as it is by auroPos.

---

```scala
body match {
  case _: Term.Ref => body
  // ...
}
```

might work:+1: I'l take a look.

Reply:
https://github.com/scalameta/scalameta/pull/2736/commits/9e77de9824a6a861de2b3458e44ed8aa10a52722 What do you think about this?